samza-core/src/test/java/org/apache/samza/clustermanager/TestContainerAllocatorWithoutHostAffinity.java

```diff
@@ -290,4 +292,53 @@ public void testRequestAllocationWithRunStreamProcessor() throws Exception {
     assertTrue(state.preferredHostRequests.get() == 0);
     spyAllocator.stop();
   }
+
+  @Test
+  public void testExpiredRequestAllocationOnAnyHost() throws Exception {
+    MockClusterResourceManager spyManager = spy(new MockClusterResourceManager(callback, state));
+    spyAllocator = Mockito.spy(
+        new ContainerAllocator(spyManager, config, state, false, Optional.empty()));
+
+    // Request Resources
+    spyAllocator.requestResources(new HashMap<String, String>() {
+      {
+        put("0", "abc");
+        put("1", "def");
+      }
+    });
+
+    spyThread = new Thread(spyAllocator);
+    // Start the container allocator thread periodic assignment
+    spyThread.start();
+
+    // Let the request expire, expiration timeout is 3 ms
+    Thread.sleep(20);
+
+    // Verify that all the request that were created as ANY_HOST host
+    // and all created requests expired
+    assertEquals(state.preferredHostRequests.get(), 0);
+    // Atleast 2 requests should expire & 2 ANY_HOST requests should be generated
+    assertTrue(state.anyHostRequests.get() >= 2);
```

Comment:
Shouldnt this be 4, there would be 2 initial requests than expire and 2 subsequent ones made in response to the expired ones?